clicache/src/DistributedSystem.hpp

```diff
@@ -122,7 +122,7 @@ namespace Apache
         /// </returns>
         static DistributedSystem^ Create(native::DistributedSystem* nativeptr);
 
-        DistributedSystem(std::unique_ptr<native::DistributedSystem> nativeptr);
+        DistributedSystem(native::DistributedSystem nativeDistributedSystem);
```

Comment:
This results in a copy of the DistributedSystem. We don't want to make a copy, in this case we would move into here and then into a `std::unique_ptr` allocated heap instance.